src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -159,18 +173,23 @@ internal override GeneratedLabelSymbol BreakLabel
 
         protected override ImmutableArray<LabelSymbol> BuildLabels()
         {
-            ArrayBuilder<LabelSymbol> labels = null;
+            // We bind the switch expression and the switch case label expressions so that the constant values can be
+            // part of the label, but we do not report any diagnostics here. Diagnostics will be reported during binding.
 
+            ArrayBuilder<LabelSymbol> labels = ArrayBuilder<LabelSymbol>.GetInstance();
+            DiagnosticBag tempDiagnosticBag = DiagnosticBag.GetInstance();
```

Comment:
Given that we always throw away these diagnostics, wouldn't it be more efficient to have a version of `DiagnosticBag` which didn't even temporarily store the values?  Essentially a bag that had a hole in the bottom.